sdk/typescript/src/signers/signer-with-provider.ts

```diff
@@ -95,20 +99,34 @@ export abstract class SignerWithProvider implements Signer {
       transaction instanceof Base64DataBuffer ||
       transaction.kind === 'bytes'
     ) {
-      const txBytes =
+      const txBytesWithTag =
         transaction instanceof Base64DataBuffer
           ? transaction
           : new Base64DataBuffer(transaction.data);
-      const sig = await this.signData(txBytes);
+      const version = await this.provider.getRpcApiVersion();
+      let dataToSign;
+      let txBytesToSubmit;
+      if (version?.major == 0 && version?.minor < 18) {
+        dataToSign = txBytesWithTag;
+        txBytesToSubmit = txBytesWithTag;
+      } else {
+        let without_tag = txBytesWithTag.getData().slice(TRANSACTION_DATA_TYPE_TAG.length);
+        const intentMessage = new Uint8Array(INTENT_BYTES.length + without_tag.length);
+        intentMessage.set(INTENT_BYTES);
+        intentMessage.set(without_tag, INTENT_BYTES.length);
+        
+        dataToSign = new Base64DataBuffer(intentMessage);
+        txBytesToSubmit = new Base64DataBuffer(without_tag);
+      }
+      const sig = await this.signData(dataToSign);
```

Comment:
here the tx_bytes submitted is without tag, the data that the sig commits to is tx data without tag, but with intent bytes 

issue here is the rpc serialize data returns tx data with tag so we need to remove it.